tabpfn_time_series/features/feature_pipeline.py

```diff
@@ -0,0 +1,576 @@
+import pandas as pd
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.pipeline import Pipeline
+import copy
+
+from typing import List, Literal, Optional, Tuple
+
+from scipy import fft
+from scipy.signal import find_peaks
+from statsmodels.tsa.stattools import acf
+
+import gluonts.time_feature
+
+
+class FeatureTransformer(BaseEstimator, TransformerMixin):
+    def __init__(self, pipeline_steps):
+        """
+        This transformer takes the steps for a pipeline as input.
+
+        Args:
+            pipeline_steps (list): A list of tuples, where each tuple contains the
+                                   name and the transformer instance, e.g.,
+                                   [('step_name', TransformerObject())].
+        """
+        self.pipeline_steps = pipeline_steps
+        # Create an unfitted template pipeline from the provided steps.
+        # We will clone this template for each group.
+        self._template_pipeline = Pipeline(steps=self.pipeline_steps)
+        self.fitted_pipelines_ = {}  # Dictionary to store a fitted pipeline for each group
+
+    def fit(self, X, y=None):
+        """
+        Fit one separate pipeline for each item_id.
+
+        This method iterates through each 'item_id' in the input DataFrame X,
+        creates a deep copy (clone) of the template pipeline, and fits that
+        copy on the data for that specific item. This ensures that each item_id
+        gets its own independently fitted pipeline without any state leaking
+        from other items.
+        """
+        # Reset fitted pipelines on each call to fit
+        self.fitted_pipelines_ = {}
+
+        grouped = X.groupby("item_id")
```

Comment:
Would this pipeline work for `RunningIndexFeatureTransformer` in `global timestamp` mode?